mantle/platform/platform.go

```diff
@@ -100,9 +100,12 @@ type Cluster interface {
 	// Name returns a unique name for the Cluster.
 	Name() string
 
-	// NewMachine creates a new Container Linux machine.
+	// NewMachine creates a new CoreOS machine.
 	NewMachine(userdata *conf.UserData) (Machine, error)
 
+	// NewMachineWithDisks creates a new CoreOS machine with additional disks.
+	NewMachineWithDisks(userdata *conf.UserData, addDisks []string) (Machine, error)
```

Comment:
Does it make sense to have a separate method for this, rather than generalizing `NewMachineWithOptions` to all platforms?

Reply:
I'd be +1 on changing this to be `NewMachineWithOptions` and having a generalized `MachineOptions` type. I really like shifting the general implementations up to the interface so we can drop explicit type checking.

I am worried that the current implementation, forcing test errors rather than skipping tests that are requesting additional resources on platforms that don't implement them yet, will be a bit obtuse to work around once tests are added to kola that make use of it.  I wonder if there might be some value in implementing a platform check + skip into the harness (similar to what's done for `--no-net`).

Reply:
I'm +1 on `NewMachineWithOptions` as well.

Reply:
This was a bit trickier than expected, because there are some options that are just specific to QEMU, like host port forwarding and pdeathsig handling. I split out a new QEMU-specific type that's a superset of the now generalized `MachineOptions` type. This does still allow us to clean up the type-checking.

Reply:
> I am worried that the current implementation, forcing test errors rather than skipping tests that are requesting additional resources on platforms that don't implement them yet, will be a bit obtuse to work around once tests are added to kola that make use of it.

Hmm, can you expand on this? Support for e.g. AWS tests that want to use `AdditionalDisks` is contingent on first adding `AdditionalDisks` support for AWS, right?

Reply:
> Hmm, can you expand on this? Support for e.g. AWS tests that want to use AdditionalDisks is contingent on first adding AdditionalDisks support for AWS, right?

Yes if the test wants to use AdditionalDisks then it requires that the platform support it. With as it was at the time I made the comment (haven't gone through the current code yet) if I say add an external test that requires additional disks and then run my external tests on qemu & aws in CI the tests for aws always fail because we don't support additional disks. What I was talking about was adding something to the harness that would check if the test needed additional disks and if so skip the test if the platform was one that doesn't have the support for adding them.